Documentation/operations/upgrade.rst

```diff
@@ -375,6 +375,9 @@ Removed Options
 
 * The long defunct and undocumented ``single-cluster-route`` flag has been removed.
 
+* The ``dnsproxy-lock-count`` and ``dnsproxy-lock-timeout`` flags no longer have
+  any effect and are deprecated. They will be removed in a future release.
```

Comment:
nit: 

```suggestion
  any effect and are deprecated. They will be removed in v1.16.
```

given that the option usage says so :smile:

Reply:
This PR is green :-) I'll fix this separately :stuck_out_tongue_winking_eye:

Reply:
https://github.com/cilium/cilium/pull/29531